airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py

```diff
@@ -199,28 +205,32 @@ def test_4xx_error_codes_http_stream(mocker, http_code):
     req.status_code = http_code
     mocker.patch.object(requests.Session, "send", return_value=req)
 
-    with pytest.raises(requests.exceptions.HTTPError):
+    with pytest.raises(MessageRepresentationAirbyteTracedErrors):
         list(stream.read_records(SyncMode.full_refresh))
 
 
 class AutoFailFalseHttpStream(StubBasicReadHttpStream):
     raise_on_http_errors = False
     max_retries = 3
-    retry_factor = 0.01
+
+    def get_error_handler(self) -> Optional[ErrorHandler]:
+        return HttpStatusErrorHandler(logging.getLogger(), max_retries=3)
 
 
 def test_raise_on_http_errors_off_429(mocker):
+    mocker.patch("time.sleep", lambda x: None)
     stream = AutoFailFalseHttpStream()
     req = requests.Response()
     req.status_code = 429
 
     mocker.patch.object(requests.Session, "send", return_value=req)
-    with pytest.raises(DefaultBackoffException, match="Request URL: https://test_base_url.com/, Response Code: 429"):
+    with pytest.raises(DefaultBackoffException, match="Too many requests"):
         list(stream.read_records(SyncMode.full_refresh))
 
 
 @pytest.mark.parametrize("status_code", [500, 501, 503, 504])
 def test_raise_on_http_errors_off_5xx(mocker, status_code):
+    mocker.patch("time.sleep", lambda x: None)
```

Comment:
Can we have this set up as part of our test config? I don't think we ever want to wait during tests

Reply:
already tried to, got an error for some tests:
![image](https://github.com/airbytehq/airbyte/assets/36314070/fba2995a-00e1-4130-94b0-5f387009bf2c)

Reply:
My understanding is that:
* for `TestMovingWindowCallRatePolicy`, we force sleep during rate limit tests and if we don't, it breaks the test. 
* for `test_backoff_strategy_max_time`, I don't understand because we already have the annotation `@pytest.mark.usefixtures("mock_sleep")` on the test method. I don't know what it does but I would assume it basically do the same thing. With this test failing though, I would assume my assumption is very bad though...

I'm fine with keeping them as such then. Thanks for checking it out!